### PR TITLE
Fix str on exception on Python 2

### LINKED ISSUE #150: UnicodeEncodeError: 'ascii' codec can't encode character u'\\u2018' in position 593: ordinal not in range(128)
Even after updating to 0.6.3 I am still seeing this exception:

```json
{
    "_ansible_parsed": false,
    "_ansible_no_log": false,
    "_ansible_delegated_vars": {
        "ansible_delegated_host": "localhost",
        "ansible_host": "localhost"
    },
    "module_stderr": "/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\n/var/lib/awx/venv/ansible/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:852: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings\n  InsecureRequestWarning)\nTraceback (most recent call last):\n  File \"/users/awx/.ansible/tmp/ansible-tmp-1592313968.07-172203014694835/AnsiballZ_azure_rm_virtualmachine.py\", line 113, in <module>\n    _ansiballz_main()\n  File \"/users/awx/.ansible/tmp/ansible-tmp-1592313968.07-172203014694835/AnsiballZ_azure_rm_virtualmachine.py\", line 105, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/users/awx/.ansible/tmp/ansible-tmp-1592313968.07-172203014694835/AnsiballZ_azure_rm_virtualmachine.py\", line 48, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_vV4S86/__main__.py\", line 1795, in <module>\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_vV4S86/__main__.py\", line 1791, in main\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_vV4S86/__main__.py\", line 771, in __init__\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_vV4S86/ansible_azure_rm_virtualmachine_payload.zip/ansible/module_utils/azure_rm_common.py\", line 308, in __init__\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_vV4S86/__main__.py\", line 1144, in exec_module\n  File \"/tmp/ansible_azure_rm_virtualmachine_payload_vV4S86/__main__.py\", line 1585, in create_or_update_vm\n  File \"/var/lib/awx/venv/ansible/lib/python2.7/site-packages/msrestazure/azure_exceptions.py\", line 192, in __str__\n    return str(self.error)\n  File \"/var/lib/awx/venv/ansible/lib/python2.7/site-packages/msrestazure/azure_exceptions.py\", line 89, in __str__\n    error_str += \"\\nMessage: {}\".format(self._message)\nUnicodeEncodeError: 'ascii' codec can't encode character u'\\u2018' in position 593: ordinal not in range(128)\n",
    "changed": false,
    "module_stdout": "",
    "rc": 1,
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
}
```

This line appears to not be catching the exception:

../msrestazure/azure_exceptions.py
```python
error_str += "\nMessage: {}".format(self._message)
```

Is there a fix that can be done to account for other characters like u'\\u2018' that are throwing this exception? This is in result of a quota in the subscription being maxed out.

### PR DESCRIPTION
**Pull request by lmazuel**
Fix #150 

The root cause is the incorrect usage of format:
```python
>>> "{}".format(u"")
''
```

Because of that on Python 2, we were loosing the uncode strings and loosing information

### CODE REVIEW

File: msrestazure/azure_exceptions.py [hunk 1]
```diff
@@ -261,6 +274,6 @@ def __init__(self, type, info):
 
     def __str__(self):
         """Cloud error message."""
-        error_str = "Type: {}".format(self.type)
-        error_str += "\nInfo: {}".format(json.dumps(self.info, indent=4))
+        error_str = u"Type: {}".format(self.type)
+        error_str += u"\nInfo: {}".format(json.dumps(self.info, indent=4))
```
Comment (johanste) [inline msrestazure/azure_exceptions.py:278]: What happens here is self.info contains non-unicode values? Shouldn't we also pass `ensure_ascii = False` to fully handle that?

**Comment by lmazuel** [comment]
@johanste by default JSON parsing will be unicode strings:
```python
Python 2.7.18 (v2.7.18:8d21aa21f2, Apr 20 2020, 13:25:05) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> json.loads('{"foo": "bar"}')
{u'foo': u'bar'}
```

Which means most (if not all) of the arguments of those formats will come from deserialized payload as unicode strings already.

File: msrestazure/azure_exceptions.py (hunk 1 above)
Comment (lmazuel) [reply msrestazure/azure_exceptions.py:278]: Good point
```python
>>> json.dumps({u'foo': u'bar'})
'{"foo": "bar"}'
>>> json.dumps({u'foo': u'bar'}, ensure_ascii=False)
u'{"foo": "bar"}'
```

Probably worth it here

Comment (lmazuel) [reply msrestazure/azure_exceptions.py:278]: Though it probably won't change a lot for a `str` usage here:
```python
>>> json.dumps({u'foo': u'ééé'}, ensure_ascii=False)
u'{"foo": "\xe9\xe9\xe9"}'
>>> json.dumps({u'foo': u'ééé'})
'{"foo": "\\u00e9\\u00e9\\u00e9"}'
```

Both are technically correct dumps of JSON

Comment (lmazuel) [reply msrestazure/azure_exceptions.py:278]: Actually on a print the first one looks better:
```python
>>> print(json.dumps({u'foo': u'ééé'}, ensure_ascii=False))
{"foo": "ééé"}
```

So I will add the ensure_ascii, thanks @johanste didn't realized that

**Pull request merged**